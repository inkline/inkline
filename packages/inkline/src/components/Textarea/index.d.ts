import { Component } from 'vue';

declare const ITextarea: Component;

export default ITextarea;
