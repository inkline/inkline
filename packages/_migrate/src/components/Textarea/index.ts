export { default as Textarea } from '@inkline/inkline/components/Textarea/Textarea.vue';
