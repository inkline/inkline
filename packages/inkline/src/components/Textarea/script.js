import IInput from '@inkline/inkline/src/components/IInput';

export default {
    name: 'ITextarea',
    extends: IInput
};
