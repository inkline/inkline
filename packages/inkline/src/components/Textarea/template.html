<div class="form-input-group" :class="classes['root']">
    <div class="form-input-prepend" v-if="$slots.prepend">
        <slot name="prepend"></slot>
    </div>

    <div class="form-input"
         :class="classes['child']"
         @mouseenter="emitMouseEnter"
         @mouseleave="emitMouseLeave">

        <span class="form-input-prefix" v-if="$slots.prefix">
            <slot name="prefix"></slot>
        </span>

        <span class="form-input-suffix" v-if="$slots.suffix || clearable">
            <i class="form-input-icon -circle" aria-label="Clear" v-show="isClearable" @click="clear">&times;</i>
            <slot name="suffix"></slot>
        </span>

        <textarea
            v-bind="$attrs"
            v-model="model"
            ref="input"
            :name="name"
            :tabindex="tabIndex"
            :disabled="isDisabled"
            :aria-disabled="isDisabled"
            :readonly="isReadonly"
            :aria-readonly="isReadonly"
            @click="emitClick"
            @focus="emitFocus"
            @blur="emitBlur"
            @change="emitChange"
            @keydown="emitKeydown"
            @keyup="emitKeyup">
        </textarea>
    </div>

    <div class="form-input-append" v-if="$slots.append">
        <slot name="append"></slot>
    </div>
</div>
