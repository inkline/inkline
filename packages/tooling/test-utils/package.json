{
    "name": "@inkline/test-utils",
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "dev": "devtools tsup watch",
        "build": "devtools tsup build",
        "lint": "devtools eslint"
    },
    "dependencies": {
        "@inkline/vue": "workspace:*"
    },
    "devDependencies": {
        "@inkline/devtools": "workspace:*",
        "@inkline/eslint-config": "workspace:*",
        "@inkline/types": "workspace:*",
        "@types/node": "catalog:",
        "@types/shelljs": "catalog:",
        "eslint": "catalog:",
        "tsx": "catalog:",
        "tsup": "catalog:",
        "typescript": "catalog:",
        "type-fest": "catalog:",
        "vue": "catalog:"
    },
    "peerDependencies": {
        "vue": ">=3.0.0"
    },
    "files": [
        "dist",
        "LICENSE"
    ],
    "types": "./dist/index.d.ts",
    "main": "./dist/index.js",
    "module": "./dist/index.cjs",
    "exports": {
        ".": {
            "types": "./dist/index.d.ts",
            "import": "./dist/index.js",
            "require": "./dist/index.cjs"
        },
        "./*": "./*"
    },
    "funding": {
        "type": "github",
        "url": "https://github.com/sponsors/alexgrozav"
    },
    "bugs": {
        "url": "https://github.com/inkline/inkline/issues"
    },
    "homepage": "https://inkline.io",
    "keywords": [
        "inkline",
        "ui",
        "ux",
        "dx",
        "library",
        "framework",
        "typescript"
    ],
    "engines": {
        "node": "*",
        "pnpm": "*"
    },
    "license": "ISC",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/inkline/inkline.git"
    }
}
