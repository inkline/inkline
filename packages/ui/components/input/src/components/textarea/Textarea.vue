<script lang="ts">
import { defineComponent, h } from 'vue';
import { Input } from '../input';
import { uid } from '@inkline/utils';

const componentName = 'Textarea';

export default defineComponent({
    name: componentName,
    components: {
        Input
    },
    inheritAttrs: true,
    props: {
        ...Input.props,
        /**
         * The unique identifier of the textarea
         * @type String
         * @default uid()
         * @name name
         */
        name: {
            type: String,
            default: () => {
                return uid('textarea');
            }
        }
    },
    emits: [
        /**
         * Event emitted for setting the modelValue
         * @type event
         * @name update:modelValue
         */
        'update:modelValue',
        /**
         * Event emitted when clearing the input element
         * @type event
         * @name clear
         */
        'clear'
    ],
    setup(props, { attrs, slots, emit }) {
        const onUpdateModelValue = (value: string) => {
            emit('update:modelValue', value);
        };

        const onClear = () => {
            emit('clear');
        };

        // Bind and re-emit input events
        const bindings = {
            onClear,
            'onUpdate:modelValue': onUpdateModelValue
        };

        return () =>
            h(
                Input,
                {
                    ...attrs,
                    ...props,
                    ...bindings,
                    type: 'textarea'
                },
                slots
            );
    }
});
</script>
