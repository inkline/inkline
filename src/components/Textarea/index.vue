<template src="./template.html" />
<style src="../Input/style.scss" lang="scss" />
<script src="./script.js" />
