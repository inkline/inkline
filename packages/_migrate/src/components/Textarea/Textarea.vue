<script lang="ts">
import { defineComponent } from 'vue';
import { Input } from '@inkline/inkline/components/Input';
import { uid } from '@inkline/utils';

const componentName = 'Textarea';

export default defineComponent({
    name: componentName,
    components: {
        Input
    },
    inheritAttrs: true,
    props: {
        ...Input.props,
        /**
         * The unique identifier of the textarea
         * @type String
         * @default uid()
         * @name name
         */
        name: {
            type: String,
            default: () => {
                return uid('textarea');
            }
        }
    },
    emits: Input.emits,
    setup(_props, { emit }) {
        function onUpdateModelValue(value: string) {
            emit('update:modelValue', value);
        }

        return {
            onUpdateModelValue
        };
    }
});
</script>

<template>
    <Input
        v-bind="{ ...$attrs, ...$props }"
        type="textarea"
        @update:model-value="onUpdateModelValue"
    >
        <template v-if="$slots.prepend" #prepend>
            <slot name="prepend" />
        </template>
        <template v-if="$slots.prefix" #prefix>
            <slot name="prefix" />
        </template>
        <template v-if="$slots.clearable" #clearable>
            <slot name="clearable" />
        </template>
        <template v-if="$slots.suffix" #suffix>
            <slot name="suffix" />
        </template>
        <template v-if="$slots.append" #append>
            <slot name="append" />
        </template>
    </Input>
</template>
