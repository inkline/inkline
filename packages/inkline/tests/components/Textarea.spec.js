import { shallowMount } from '@vue/test-utils';
import Textarea from '@inkline/inkline/src/components/Textarea';

describe('Components', () => {
    describe('Textarea', () => {
        let wrapper;

        beforeEach(() => {
            wrapper = shallowMount(Textarea);
        });

        it('should be named correctly', () => {
            expect(Textarea.name).toEqual('ITextarea');
        });

        it('should render correctly', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
