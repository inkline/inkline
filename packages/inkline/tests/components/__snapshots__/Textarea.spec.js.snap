// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Textarea should render correctly 1`] = `
<div class="form-input-group">
  <!---->
  <div class="form-input">
    <!---->
    <!----> <textarea tabindex="1"></textarea></div>
  <!---->
</div>
`;
